### PR TITLE
feet: actually allow setup function to be a coroutine

### LINKED ISSUE #468: Async setup
For ext.commands allow setup to be async

### PR DESCRIPTION
**Pull request by MaskDuck**
## Summary
close #468 
allow the setup func to be a coworoutine

## Checklist

- [ ] If code changes were made then they have been tested. (if anyone are free plz test)
    - [ ] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/ext/commands/bot.py [hunk 1]
```diff
@@ -698,7 +698,10 @@ def _load_from_module_spec(
 
         extras = extras or {}
         try:
-            setup(self, **extras)
+            if inspect.iscoroutinefunction(setup):
+                asyncio.run(setup(self, **extras))
```
Comment (Skelmis) [inline nextcord/ext/commands/bot.py:702]: Asyncio.run is used as a program entry point, and should ideally only be called once during a programs lifecycle. Therefore using it for each cog is not the go

https://docs.python.org/3/library/asyncio-task.html#asyncio.run

**Review by Skelmis — CHANGES_REQUESTED** [review]

File: nextcord/ext/commands/bot.py (hunk 1 above)
Comment (MaskDuck) [reply nextcord/ext/commands/bot.py:702]: how do you think about making _load_from_module_specs a coroutine?

**Comment by MaskDuck** [comment]
if someone have permissions can you mark this pr as draft thanks

**Comment by MaskDuck** [comment]
sorry didnt know it was a thing

**Comment by MaskDuck** [comment]
uh i have an idea how to implement it now

**Comment by MaskDuck** [comment]
uh fork again ig

**Pull request closed without merging**